src/System.Net.Http/src/System/Net/Http/SocketsHttpHandler/HttpEnvironmentProxy.cs

```diff
@@ -283,7 +283,10 @@ public ICredentials Credentials
             {
                 return _credentials;
             }
-            set { throw new NotSupportedException(); }
+            set
+            {
```

Comment:
Why did you add support for set here (and elsewhere)?

I'm not sure what the right behavior here is, I just want to understand the rationale.

Do we support setting this on Framework when you call GetSystemWebProxy?

Reply:
> Do we support setting this on Framework when you call GetSystemWebProxy?

Yup.  `GetSystemWebProxy` just returns a `WebProxy` (well, a wrapper around one that just delegates to it)
https://referencesource.microsoft.com/#System/net/System/Net/WebRequest.cs,1092
and WebProxy.set_Credentials just stores them:
https://referencesource.microsoft.com/#System/net/System/Net/webproxy.cs,227

> Why did you add support for set here (and elsewhere)?

I initially hadn't, and then a bunch of WebRequest tests started failing because they expected to be able to.  That led me to look around at what's done in netfx and in other proxy implementations, and they pretty much all just support roundtripping the set credentials.  I didn't find any that threw NotSupportedException like these ones are doing.  And the docs don't say anything about NotSupportedException.  So, it seemed like more of a pit of failure to throw than to just let the set value roundtrip.

Reply:
Ok, thanks.